app/model/Travel/TravelService.php

```diff
@@ -183,6 +183,22 @@ public function addTravel(int $commandId, string $type, Date $date, string $star
         $this->commands->save($command);
     }
 
+    public function duplicateTravel(int $commandId, int $travelId) : void
+    {
+        $command       = $this->commands->find($commandId);
+        $travel        = $this->getTravel($commandId, $travelId);
+        $details       = $travel->getDetails();
+        $transportType = $this->types->find($details->getTransportType());
+
+        if ($transportType->hasFuel()) {
```

Comment:
Tady bych to řešil přes shallow copy a rovnou insert nove, tedy v Travel udělat metodu duplicate(), ktera uděla novou instaci bez id. Název Metody možná ještě upraví @fmasa

Reply:
Bylo by pak potřeba udělat setter na id a zajistit, aby bylo stejné v poli travels a tom objektu, teď to za mě řeší ty add(Vehicle|Transport)Travel